clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -579,11 +629,34 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
             ListOffsetResponse.PartitionData partitionData = listOffsetResponse.responseData().get(topicPartition);
             Errors error = Errors.forCode(partitionData.errorCode);
             if (error == Errors.NONE) {
-                OffsetAndTimestamp offsetAndTimestamp = null;
-                if (partitionData.offset != ListOffsetResponse.UNKNOWN_OFFSET)
-                    offsetAndTimestamp = new OffsetAndTimestamp(partitionData.offset, partitionData.timestamp);
-                log.debug("Fetched {} for partition {}", offsetAndTimestamp, topicPartition);
-                timestampOffsetMap.put(topicPartition, offsetAndTimestamp);
+                if (partitionData.offsets != null) {
+                    // Handle v0 response
+                    long offset;
+                    if (partitionData.offsets.size() > 1) {
+                        throw new IllegalStateException("Unexpected partitionData response of length " +
+                                partitionData.offsets.size());
+                    } else if (partitionData.offsets.isEmpty()) {
+                        offset = ListOffsetResponse.UNKNOWN_OFFSET;
+                    } else {
+                        offset = partitionData.offsets.get(0);
+                    }
+                    log.debug("handleListResponse: handling v0 response for " + topicPartition +
```

Comment:
nit: string interpolation is preferred. another one below. Also, not sure about the convention of including the function name. Wouldn't the logger do that for us?